### PR TITLE
Refactoring/azure/monitor

### PR DESCRIPTION
**Pull request by misg**
This PR aims to migrate Azure Monitor service to the new architecture, as part of #183.

### CODE REVIEW

File: ScoutSuite/providers/azure/resources/monitor/activity_logs.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        # TODO: build that facade somewhere else:
+        facade = MonitorFacade(credentials.credentials, credentials.subscription_id)
+
+        self['activity_logs'] = {}
+        self['activity_logs']['storage_accounts'] = {}
+
+        for raw_log in await facade.get_activity_logs():
+            self._parse(raw_log)
+
+    def _parse(self, raw_log):
```
Comment (Aboisier) [inline ScoutSuite/providers/azure/resources/monitor/activity_logs.py]:
```suggestion
    def _parse_log(self, raw_log):
```

File: ScoutSuite/providers/azure/resources/monitor/activity_logs.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    async def fetch_all(self, credentials, **kwargs):
+        # TODO: build that facade somewhere else:
+        facade = MonitorFacade(credentials.credentials, credentials.subscription_id)
+
+        self['activity_logs'] = {}
+        self['activity_logs']['storage_accounts'] = {}
+
+        for raw_log in await facade.get_activity_logs():
+            self._parse(raw_log)
```
Comment (Aboisier) [inline ScoutSuite/providers/azure/resources/monitor/activity_logs.py]:
```suggestion
            self._parse_log(raw_log)
```

File: ScoutSuite/providers/azure/resources/monitor/activity_logs.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    def _parse_storage_account_log(self, raw_log):
+            storage_account_id = get_non_provider_id(raw_log.resource_id.lower())
+
+            if storage_account_id not in self['activity_logs']['storage_accounts']:
+                self['activity_logs']['storage_accounts'][storage_account_id] =\
+                    {'access_keys_rotated': False}
+
+            if raw_log.operation_name.value == 'Microsoft.Storage/storageAccounts/regenerateKey/action':
+                self['activity_logs']['storage_accounts'][storage_account_id]['access_keys_rotated'] = True
```
Comment (Aboisier) [inline ScoutSuite/providers/azure/resources/monitor/activity_logs.py]: The value is `True` or `None`? Why not `True`/`False`?

Comment (misg) [reply ScoutSuite/providers/azure/resources/monitor/activity_logs.py]: ? The value is True or False, I'm setting it to False by default, one line above.

Comment (Aboisier) [reply ScoutSuite/providers/azure/resources/monitor/activity_logs.py]: My bad, I didn't notice the previous line lel

**Pull request merged**